URL: https://github.com/TDFKAOlli/knock/commit/4a5e12b2c01bd13d8d5ece5c8cbc6139ec2de765
Summary: Fixes issue with uninitialized parameters in door structure for tcpflags
----
--- a/src/knockd.c.orig		2019-02-26 06:55:13.667005918 +0000
+++ b/src/knockd.c		2019-02-26 06:56:55.842630851 +0000
@@ -509,6 +509,12 @@ int parseconfig(char *configfile)
 				door->start_command = NULL;
 				door->cmd_timeout = CMD_TIMEOUT; /* default command timeout (seconds) */
 				door->stop_command = NULL;
+				door->flag_fin = DONT_CARE;
+				door->flag_syn = DONT_CARE;
+				door->flag_rst = DONT_CARE;
+				door->flag_psh = DONT_CARE;
+				door->flag_ack = DONT_CARE;
+				door->flag_urg = DONT_CARE;
 				door->one_time_sequences_fd = NULL;
 				door->pcap_filter_exp = NULL;
 				doors = list_add(doors, door);
